common/src/main/java/com/wynntils/models/beacons/type/Beacon.java

```diff
@@ -4,6 +4,41 @@
  */
 package com.wynntils.models.beacons.type;
 
-import net.minecraft.world.phys.Vec3;
+import com.wynntils.utils.mc.type.PreciseLocation;
+import java.util.Objects;
 
-public record Beacon(Vec3 position, BeaconKind beaconKind) {}
+public final class Beacon<T extends BeaconKind> {
```

Comment:
Why did you convert this from a record? Is it because records cannot be templates? (I honestly have no clue if this is the case or not)